File: Net_Framework/CS/GrowthWare.WebApplication/UserControls/RolesControl.ascx.cs
using System;
using System.Collections;
using System.Web.UI;

namespace GrowthWare.WebApplication.UserControls
{
    public partial class RolesControl : UserControl
    {
        #region Public Properties
        public ArrayList AllRoles;
        public Array SelectedViewRoles;
        public Array SelectedAddRoles;
        public Array SelectedEditRoles;
        public Array SelectedDeleteRoles;
        public bool ViewRolesChanged
        {
            get { return ctlViewRoles.Changed; }
        }
        public string ViewRoles
        {
            get { return ctlViewRoles.SelectedState; }
        }
        public string AddRoles
        {
            get { return ctlAddRoles.SelectedState; }
        }
        public bool AddRolesChanged
        {
            get { return ctlAddRoles.Changed; }
        }
        public string EditRoles
        {
            get { return ctlEditRoles.SelectedState; }
        }
        public bool EditRolesChanged
        {
            get { return ctlEditRoles.Changed; }
        }
        public string DeleteRoles
        {
            get { return ctlDeleteRoles.SelectedState; }
        }
        public bool DeleteRolesChanged
        {
            get { return ctlDeleteRoles.Changed; }
        }
        #endregion

        #region Page Event Handlers
        protected void Page_Load(object sender, System.EventArgs e)
        {
            ctlViewRoles.DataSource = AllRoles;
            ctlAddRoles.DataSource = AllRoles;
            ctlEditRoles.DataSource = AllRoles;
            ctlDeleteRoles.DataSource = AllRoles;
            if ((SelectedViewRoles != null)) ctlViewRoles.SelectedItems = (string[])SelectedViewRoles;
            if ((SelectedAddRoles != null)) ctlAddRoles.SelectedItems = (string[])SelectedAddRoles;
            if ((SelectedEditRoles != null)) ctlEditRoles.SelectedItems = (string[])SelectedEditRoles;
            if ((SelectedDeleteRoles != null)) ctlDeleteRoles.SelectedItems = (string[])SelectedDeleteRoles;
            DataBind();
        }
        #endregion
    }
}

File: Net_Framework/VB/GrowthWare.WebApplication/UserControls/RolesControl.ascx.vb
Public Class RolesControl
    Inherits System.Web.UI.UserControl

    Public AllRoles As ArrayList
    Public SelectedViewRoles As Array
    Public SelectedAddRoles As Array
    Public SelectedEditRoles As Array
    Public SelectedDeleteRoles As Array

    Public ReadOnly Property ViewRolesChanged() As Boolean
        Get
            Return ctlViewRoles.Changed
        End Get
    End Property

    Public ReadOnly Property ViewRoles() As String
        Get
            Return ctlViewRoles.SelectedState
        End Get
    End Property

    Public ReadOnly Property AddRoles() As String
        Get
            Return ctlAddRoles.SelectedState
        End Get
    End Property

    Public ReadOnly Property AddRolesChanged() As Boolean
        Get
            Return ctlAddRoles.Changed
        End Get
    End Property

    Public ReadOnly Property EditRoles() As String
        Get
            Return ctlEditRoles.SelectedState
        End Get
    End Property

    Public ReadOnly Property EditRolesChanged() As Boolean
        Get
            Return ctlEditRoles.Changed
        End Get
    End Property

    Public ReadOnly Property DeleteRoles() As String
        Get
            Return ctlDeleteRoles.SelectedState
        End Get
    End Property

    Public ReadOnly Property DeleteRolesChanged() As Boolean
        Get
            Return ctlDeleteRoles.Changed
        End Get
    End Property

    Protected Sub Page_Load(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.Load
        ctlViewRoles.DataSource = AllRoles
        ctlAddRoles.DataSource = AllRoles
        ctlEditRoles.DataSource = AllRoles
        ctlDeleteRoles.DataSource = AllRoles
        If Not SelectedViewRoles Is Nothing Then ctlViewRoles.SelectedItems = SelectedViewRoles
        If Not SelectedAddRoles Is Nothing Then ctlAddRoles.SelectedItems = SelectedAddRoles
        If Not SelectedEditRoles Is Nothing Then ctlEditRoles.SelectedItems = SelectedEditRoles
        If Not SelectedDeleteRoles Is Nothing Then ctlDeleteRoles.SelectedItems = SelectedDeleteRoles
        DataBind()
    End Sub

End Class